fuel-vm/src/interpreter/executors/instruction.rs

```diff
@@ -80,8 +80,15 @@ where
             }
         }
 
-        self.instruction_inner(raw.into())
-            .map_err(|e| InterpreterError::from_runtime(e, raw.into()))
+        let result = self
```

Comment:
I think it will be more beneficial if, on the creation of the interpreter, we will pass some object that implements traits like `Tracer` or something.

```rust
trait Tracer<VM> {
    fn trace_instruction_before_execution(
        &mut self,
        instruction: &Intstuction,
        vm: &mut VM,
    );

    fn trace_instruction_after_execution(
        &mut self,
        instruction: &Intstuction,
        vm: &mut VM,
    );
}
```

In this case, we will have more flexibility in the implementation and can define logic to get traces on the `fuel-core` side. In the case of tooling, they can maybe create a local debugger. In the case of indexers, they can index something inside of these functions. 

Of course, this solution means that we need to have getters for all internal fields, including `MemoryInstance`.

Reply:
I think nobody is actually going to use this, and it's an unnecessary complicaton. But it would be nice to be wrong about this, and it's not that much extra complication.

Fortunately we already support almost all fields this needs, as ECAL support is also done like this.